client/client.go

```diff
@@ -743,7 +749,9 @@ func (c *Client) Reload(newConfig *config.Config) error {
 	}
 
 	if shouldReloadTLS {
-		return c.reloadTLSConnections(newConfig.TLSConfig)
+		if err := c.reloadTLSConnections(newConfig.TLSConfig); err != nil {
+			return err
+		}
 	}
 
 	c.fingerprintManager.Reload()
```

Comment:
Unrelated minor fix. In theory if an operator was updating tls settings and something that affected fingerprinting in one fell swoop, we could have missed whatever affected fingerprinting.